float64 mixing_factor
float64[] mean
float64[] covariance